### PR TITLE
feat(field-label): add field label pattern

### LINKED ISSUE #475: Add Spectrum Field Label Pattern
### Expected Behaviour
https://opensource.adobe.com/spectrum-css/components/fieldlabel/ should be available in the library.

### Actual Behaviour
Missing

### The future
image

### PR DESCRIPTION
**Pull request by Westbrook**
## Description
- add field label pattern
- update docs site to use this pattern rather than custom code
- correct an issue in stealing focus after an overlay is closed

Preview available at: https://5fc853af5b61815cd54df3fd--spectrum-web-components.netlify.app

## Related Issue
fixed #475 

## Motivation and Context
Labels are an important accessibility and usability tool.

## How Has This Been Tested?
- unit tests
- visual regression tests
- manual integration tests

## Screenshots (if appropriate):
image
image
image
image

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist:
- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by jnurthen** [comment]
Clicking the label should move focus to the associated form element. This works fine for the sp-textfield but doesn't seem to work for the sp-dropdown

**Comment by Westbrook** [comment]
@jnurthen thanks for checking in. Can you confirm a couple of things for me? 

- browser/os/version
- if you "click the label" and then "press spacebar" does it act as expected with the `sp-dropdown`?

If yes to the spacebar, then this has to do with "click" not triggering the heuristic for `:focus-visible`. Should we be attempting to hack that in this case? If so, I can share this information with the Spectrum team so that this design/functionality can be perpetuated across other surfaces.

**Comment by jnurthen** [comment]
This occurred on both Safari & Chrome (Current production versions) on MacOS 10.15.7
You are correct that focus does indeed appear to be there - but the ring is not being drawn.

react-spectrum ( https://react-spectrum.adobe.com/react-spectrum/Picker.html )  does draw the focus ring in this case and I suggest that SWC should do the same.

**Comment by Westbrook** [comment]
Haha. Gotta love the way the web has to motivate entire swaths of designers, devs and product owners to get an accessible feature into the platform only to work around its presence later.

Thanks for the clarification. I'll get that worked out and added to the PR.

I'll still bring this up with Spectrum design/CSS as the `quite` picker does this by default and there should really be a less bumpy design API here. Always a little something more!

File: packages/field-label/src/FieldLabel.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            </label>
+        `;
+    }
+
+    protected firstUpdated(changes: PropertyValues): void {
+        super.firstUpdated(changes);
+        if (!this.hasAttribute('id')) {
+            this.setAttribute(
+                'id',
+                `sp-field-label-${FieldLabel.instanceCount++}`
```
Comment (adixon-adobe) [inline packages/field-label/src/FieldLabel.ts:114]: this is the sort of code that will cause subtle conflicts if multiple `sp-field-label` classes can be instantiated, or if this is subclassed. Can you use the tag name directly here? That'd at least give some safety here.

**Review by adixon-adobe — COMMENTED** [review]
I don't have time to git this a really deep review, but this looks good overall.

I've got one comment that could be improved now (or later). I don't think you need to block this on it though.

File: packages/field-label/src/FieldLabel.ts (hunk 1 above)
Comment (Westbrook) [reply packages/field-label/src/FieldLabel.ts]: Great idea. Updated!

**Comment by Westbrook** [comment]
Expanded on the tests here, and really hope we can get this reviewed/merged soon!

**Review by najikahalsema — APPROVED** [review]
Looks good to me!

**Pull request merged**